sys/mips/cheri/cheri.c

```diff
@@ -157,15 +128,9 @@ void
 cheri_serialize(struct cheri_serial *csp, void * __capability cap)
 {
 
-#if CHERICAP_SIZE == 16
 	csp->cs_storage = 3;
 	csp->cs_typebits = 16;
 	csp->cs_permbits = 23;
-#else /* CHERICAP_SIZE == 32 */
-	csp->cs_storage = 4;
-	csp->cs_typebits = 24;
-	csp->cs_permbits = 31;
-#endif
 
 	KASSERT(csp != NULL, ("Can't serialize to a NULL pointer"));
 	if (cap == NULL) {
```

Comment:
This entire function seems pointless? Why not just `memcpy` the cap? If it's tagged then it's been correctly derived...

Reply:
IIRC the idea was to keep the format the same between 128 and 256bits, but there really is no point with everything being cheri concentrate.  In fact, most of it's consumers don't make any sense any more since we're not handling CCall and CReturn in the kernel anymore.